src/cogs/gamble.py

```diff
@@ -0,0 +1,152 @@
+import random
+import time
+
+from discord.ext import commands
+import discord
+from discord.ext.commands import Context
+from discord import Member
+from datetime import datetime
+
+from src.util.user_database import *
+
+
+daily_amount = 10
+
+class Gamble(commands.Cog, name="Gamble"):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @commands.Cog.listener()
+    async def on_command_error(self, ctx: Context, error):
+        try:
+            if isinstance(error, commands.MissingRequiredArgument):
+                await ctx.send('Invalid input.')
+            if isinstance(error, commands.MissingPermissions):
+                await ctx.send('You do not have permission to use this command.')
+            if isinstance(error, commands.BadUnionArgument):
+                await ctx.send('Invalid input.')
+        except Exception as error:
+            print(error)
+
+    @commands.hybrid_command(with_app_command=True)
+    async def bet(self, ctx: Context, amount_to_bet: int):
+        """
+        Place a bet (50/50). Enter the amount to bet.
+        """
+        user_database_check_if_user_exists_otherwise_add(ctx.author.id)
+
+        if amount_to_bet == None:
+            await ctx.send('You need to enter the amount to bet.', ephemeral=True)
+            return
+    
+        amount_in_wallet = user_database_get('money', ctx.author.id)
+
+        if amount_to_bet <= 0:
+            await ctx.send('You must enter a positive number greater than 0.', ephemeral=True)
+            return
+        if amount_to_bet > amount_in_wallet:
+            await ctx.send('You don\'t have enough money.', ephemeral=True)
+            return
+
+        won = random.randint(1, 2) == 1
+
+        # Update wallet.
+        amount_result = amount_in_wallet + (amount_to_bet if won else -amount_to_bet)
+        user_database_set('money', amount_result, ctx.author.id)
+
+        if won:
+            embed = discord.Embed(title='You won!', color=discord.Color.green())
+            embed.add_field(name=f'{amount_to_bet}$ has been added to your wallet', value=f'You have {amount_result}$')
+            await ctx.send(embed=embed)
+        else:
+            embed = discord.Embed(title='You lost.', color=discord.Color.red())
+            embed.add_field(name=f'{amount_to_bet}$ has been removed from your wallet', value=f'You have {amount_result}$')
+            await ctx.send(embed=embed)
+    
+    @commands.hybrid_command(with_app_command=True)
+    async def wallet(self, ctx: Context, user: Member = None):
+        """
+        Check your wallet. (Optional) Enter another user's ID to view their wallet.
+        """
+        if user is None:
+            user = ctx.author
+
+        user_database_check_if_user_exists_otherwise_add(user.id)
+
+        embed = discord.Embed(title=f'{user.display_name}\'s wallet', color=discord.Color.blurple())
+        embed.set_thumbnail(url='https://cdn.iconscout.com/icon/free/png-512/free-wallet-588-456730.png')
```

Comment:
What's the license for that image? If you want we can add it as emoji or sticker so we don't have our bot link to something off-platform that's also entirely out of our control